### ISSUE TITLE
smslib lightweight?

### ISSUE THREAD

**Issue by glend1** [tdelenikas/smslib#11]
smslib is a great looking tool, however, for my needs its kinda slow to connect i believe this is due to all of the unnecessary stuff it does, like creating a http server, and a dozen different threads and having code run on those threads. these things seem unnecessary to me in most cases. could a version of smslib be distributed with these things stripped out? maybe it is due to the singleton nature of this application, and maybe its just not possible. 

i mention this because i am looking for a library to send sms messages as part of a larger application, my application will be sharing the gsm modem with other applications and having smslib running efficiently means keeping the connection open. i cant keep other applications from connecting to it.

**Comment by tdelenikas** [comment]
Hi,

Do you mean that you need to frequently start/stop smslib, so you require fast startup times?

I don't know how much faster it could be... The weak link here is the modem communication which is tens of times slower than all the threads you are mentioning.

**Comment by glend1** [comment]
yes, but at the same time i may not need to send a text for days or weeks.

Looking at the documentation for v4 and v3 there are a hell of a lot of dependencies which really shouldn't be necessary for sending AT commands over TCP/IP or communicating through a web service. Do you agree?

What i am interested in is a core library. Something that will allow me to **just** send texts through a java library.

This is because for the application i am building i do not need the HTTP server i also don't need an always listening service. I need a minimal solution, an example might be;

``` java
    SmsLib example = new SmsLib(host, port);
    example.setNumber(number);
    example.setMessage(message);
    example.send();
    example.close();
```

**Comment by ghost** [comment]
+1
